usdt-impl/src/common.rs

```diff
@@ -117,26 +117,28 @@ fn shared_slice_elem_type(reference: &syn::TypeReference) -> Option<&syn::Type>
     }
 }
 
+// x86_64 passes the first 6 arguments in registers, with the rest on the stack.
+// We limit this to 6 arguments in all cases for now, as handling those stack
+// arguments would be challenging with the current `asm!` macro implementation.
+#[cfg(target_arch = "x86_64")]
+pub(crate) const ABI_REGS: [&str; 6] = ["rdi", "rsi", "rdx", "rcx", "r8", "r9"];
+#[cfg(target_arch = "aarch64")]
+pub(crate) const ABI_REGS: [&str; 6] = ["x0", "x1", "x2", "x3", "x4", "x5"];
+#[cfg(not(any(target_arch = "aarch64", target_arch = "x86_64")))]
+compile_error!("USDT only supports x86_64 and ARM64 architectures");
+
 // Return code to destructure a probe arguments into identifiers, and to pass those to ASM
 // registers.
 pub fn construct_probe_args(types: &[DataType]) -> (TokenStream, TokenStream) {
-    // x86_64 passes the first 6 arguments in registers, with the rest on the stack.
-    // We limit this to 6 arguments in all cases for now, as handling those stack
-    // arguments would be challenging with the current `asm!` macro implementation.
-    #[cfg(target_arch = "x86_64")]
-    let abi_regs = ["rdi", "rsi", "rdx", "rcx", "r8", "r9"];
-    #[cfg(target_arch = "aarch64")]
-    let abi_regs = ["x0", "x1", "x2", "x3", "x4", "x5"];
-    #[cfg(not(any(target_arch = "aarch64", target_arch = "x86_64")))]
-    compile_error!("USDT only supports x86_64 and ARM64 architectures");
+    let abi_regs = &ABI_REGS;
```

Comment:
I think we can just inline this into the `.zip()` call, as `&ABI_REGS`.

Reply:
Leftover code from when I originally reused the ABI_REGS in the argument handling. Reverted these changes.